Source/WebKit/UIProcess/API/Cocoa/WKWebsiteDataStorePrivate.h

```diff
@@ -124,6 +124,12 @@ typedef NS_OPTIONS(NSUInteger, _WKWebsiteDataStoreFetchOptions) {
 -(void)_originDirectoryForTesting:(NSURL *)origin topOrigin:(NSURL *)topOrigin type:(NSString *)dataType completionHandler:(void(^)(NSString *))completionHandler WK_API_AVAILABLE(macos(13.0), ios(16.0));
 -(void)_setBackupExclusionPeriodForTesting:(double)seconds completionHandler:(void(^)(void))completionHandler WK_API_AVAILABLE(macos(13.0), ios(16.0));
 
+-(void)_getAllBackgroundFetchIdentifiers:(void(^)(NSArray<NSString *> *identifiers))completionHandler WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
+-(void)_abortBackgroundFetch:(NSString *)identifier completionHandler:(void(^)(void))completionHandler WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
+-(void)_pauseBackgroundFetch:(NSString *)identifier completionHandler:(void(^)(void))completionHandler WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
+-(void)_resumeBackgroundFetch:(NSString *)identifier completionHandler:(void(^)(void))completionHandler WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
+-(void)_clickBackgroundFetch:(NSString *)identifier completionHandler:(void(^)(void))completionHandler WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA));
```

Comment:
Looking at the implementation of these functions in network process, it looks like they could fail. Do we want to return a boolean or error to the client so that they know the operation does not succeed and take action?

Reply:
We do not do this for some other APIs so we have the option. For abort/pause/resume, the app can detect whether it succeeds.
We might want to add it for click though as they can take some counter measures.
I'll do this in a follow-up.
We should also make them nullable probably, will change this.